ports/samd/machine_uart.c

```diff
@@ -479,12 +486,18 @@ static void mp_machine_uart_init_helper(machine_uart_obj_t *self, size_t n_args,
 }
 
 static mp_obj_t mp_machine_uart_make_new(const mp_obj_type_t *type, size_t n_args, size_t n_kw, const mp_obj_t *args) {
-    mp_arg_check_num(n_args, n_kw, 1, MP_OBJ_FUN_ARGS_MAX, true);
+    mp_arg_check_num(n_args, n_kw, 0, MP_OBJ_FUN_ARGS_MAX, true);
 
     // Get UART bus.
-    int uart_id = mp_obj_get_int(args[0]);
+    int uart_id = MICROPY_HW_DEFAULT_UART_ID;
+    int arg_offset = 0;
+
+    if (n_args > 0 && mp_obj_get_int(args[0]) <= SERCOM_INST_NUM) {
+        uart_id = mp_obj_get_int(args[0]);
+        arg_offset = 1;
+    }
     if (uart_id < 0 || uart_id > SERCOM_INST_NUM) {
-        mp_raise_msg_varg(&mp_type_ValueError, MP_ERROR_TEXT("UART(%d) doesn't exist"), uart_id);
+        mp_raise_msg_varg(&mp_type_ValueError, MP_ERROR_TEXT("UART(%d) does not exist"), uart_id);
```

Comment:
Please keep this unchanged.  The word `doesn't` is slightly smaller in code size and we use contracted words in all other error messages (or at least that's the aim).

Reply:
OK.